src/v/rpc/test/rpc_gen_cycling_test.cc

```diff
@@ -454,3 +454,64 @@ FIXTURE_TEST(corrupted_data_at_server, rpc_integration_fixture) {
         BOOST_REQUIRE_EQUAL(echo_resp_new.value().data.str, "testing...");
     }
 }
+
+FIXTURE_TEST(version_not_supported, rpc_integration_fixture) {
+    configure_server();
+    register_services();
+    start_server();
+
+    rpc::transport t(client_config());
+    t.connect(model::no_timeout).get();
+    auto client = echo::echo_client_protocol(t);
+
+    const auto check_unsupported = [&] {
+        auto f = t.send_typed_versioned<echo::echo_req, echo::echo_resp>(
+          echo::echo_req{.str = "testing..."},
+          960598415,
+          rpc::client_opts(rpc::no_timeout),
+          rpc::transport_version::invalid);
```

Comment:
Do we expect `::invalid` to be used anywhere except test? Would `::max + 1` have worked just as well?

Reply:
yeh, invalid is just for testing. good point. I'll remove it from the enum to keep it lean.

Reply:
i think we need to leave `invalid` in because `enum class transport_version` is a strong type. in order to pass in max+1 at this location would mean adding some hacky workarounds for the enum value, unless i'm missing a trick that's available to us.

i added a comment however to the enum stating that invalid is for testing, and it shouldn't be considered a reserved value in the protocol, and that if it were ever needed to use those bits one could fix up the code to create an invalid verison scenario in some other way.

Reply:
We could do something like `static_cast<uint8_t>(::max) + 1`, though not sure how you feel about its hackiness. Casting always has a smell IMO, but it might be justified here

Reply:
Ah nevermind, the input here is an enum class and that'd probably not work... Let's proceed as is then

Reply:
yeh i think you can technically force the integers into the strong enum class value, but then you lose all the type safety. i think adding a comment in this case is the way to go. i also changed it from invalid to unsupported with a value of max+1 to make its semantics clearer.